### PR TITLE
workflow now makes pr instead of pushing to master

### LINKED ISSUE #135: [Bug]: Release workflows should make PRs instead of pushing to `master`
## Expected behavior

The `next-release-minor.yml` workflow should pass

## Describe the bug

The `next-release-minor.yml` workflow fails becasue it tries to push file changes directly to `master`. This causes the workflow to fail since `master` is a protected branch and changes requires an approving review from users with write access.

## To Reproduce
1. [Look at this action report](https://github.com/arfc/saltproc/runs/5237650128?check_suite_focus=true)

## System Specifications:
 
 - Software version: SaltProc v0.3.0
 - Subsystem: Github actions

## How can this issue be closed?
This issue can be closed when the `next-release-minor` workflow is changed to do the following.

Create a PR with the following changes:
- A release notes file for the new version
- Update the version number(s) in `version.py`
- Remove the "in-development" note from the release notes for the current release.

### PR DESCRIPTION
**Pull request by yardasol**
## Summary of changes

This PR addresses #135. Specifically this PR makes the `next-release-minor.yml` workflow create a new PR instead of pushing directly to master.

Do not merge until #138 is merged

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Required for Merging
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] My change is a source code change
  - [ ] I have added/modified tests to cover my changes
  - [ ] I have explained why my change does not require new/modified tests 
- [x] All new and existing tests passed.
  - [x] CI tests pass
  - [x] Local tests pass (including Serpent2 integration tests)
- [x] ~I have recorded my changes in the changelog for the upcoming release~ n/a

## Associated Issues and PRs

- Issue: closes #135

## Associated Developers

- Dev: @munkm 

## Checklist for Reviewers

Reviewers should use [this link](https://arfc.github.io/manual/guides/pull_requests) to get to the 
Review Checklist before they begin their review.

### CODE REVIEW

File: .github/workflows/next-release-minor.yml
Comment (yardasol) [reply .github/workflows/next-release-minor.yml:97]: Good suggestion!

File: .github/workflows/next-release-minor.yml
Comment (yardasol) [reply .github/workflows/next-release-minor.yml:98]: Okay. How about something like "Update `version.py` and create release notes for ${{ env.NEW_RELEASE_VERSION }}"?

File: .github/workflows/next-release-minor.yml
Comment (yardasol) [reply .github/workflows/next-release-minor.yml:99]: I was definitely having trouble articulating this clearly. The PR created by (1) should be the first PR that gets merges as part of development on version `NEW_RELEASE_VERSION` of the software. So I think I could make the body something like the following:
"This is an automated PR that updates `version.py` and adds release notes for version ${{ env.NEW_RELEASE VERSION }}. This should be the first PR merged towards the full release of ${{ env.NEW_RELEASE_VERSION }}"

Comment (yardasol) [reply .github/workflows/next-release-minor.yml:99]: I do feel like I'm having trouble getting the core of the idea in a good sentence structure, so any suggestion are greatly appreciated!

File: .github/workflows/next-release-minor.yml
Comment (yardasol) [reply .github/workflows/next-release-minor.yml:108]: Again, I was having trouble articulating my ideas when I wrote this, so that name definitely isn't as clear as I want it to be. The PR created by (3) changes `_version_extra` in `version.py` from `'dev'` to `'0'` to indicate the full release version. This should be the last PR merged before publishing `NEW_RELEASE_VERSION`, and in fact, the release should be published immediately after the PR is merged.

**Comment by yardasol** [comment]
Hi @munkm, thanks for your feedback. I definitely think I could have said a few things more clearly. I've responded to some of your in-line comments and I hope they provide some answers.

To respond to your question: 

> Would you also mind explaining the process from (1) Create PR for initial new version changes --> (2) Advance changes for publishing the new release --> and (3) Create advance PR final new version changes? Is (3) always dependent on 1 and 2? Does it depend on 1 being merged? what if 3 is merged before 1?

I explained the purpose of (1) and (3) in my responses to your inline comments above, but in summary:
- (1) creates the first PR we merge to start development on the new version of the software
- (3) creates the last PR we merge before publishing the full release of the new version of the software.

Because of this, (3) should never be merged before (1). I can specify this more explicitly in the body.

(2) simply applies the changes contained in the PR created by (3), so yes, (3) is dependent on (2).

I hope that answers your questions.

File: .github/workflows/next-release-minor.yml
Comment (munkm) [inline .github/workflows/next-release-minor.yml:130]: I *love* that you have labels auto added!!! 😍

**Review by munkm — COMMENTED** [review]
Thanks for the changes @yardasol! Your explanations were really helpful and I think the PR body text changes you've made make the difference between (1) and (3) clear. 

I see now that (3) uses the changes from (2) also. Thank you for clarifying that a bit. 

One thing I'm curious about is it looks like (1) (2) and (3) will all happen when this workflow is triggered. How will (2) and (3) get updated between the commits that get merged after (1) is merged and when you're ready for (3) to happen? Also, is (2) supposed to happen right after (1) or right before (3)?

**Comment by yardasol** [comment]
Add a sentence to the body instructing the Release manager to rebase (3) on top of the current commit in `master`

**Review by munkm — APPROVED** [review]

**Comment by yardasol** [comment]
@munkm thanks for the review! I don't want to self merge, so could you merge this after #138 gets merged?

**Pull request merged**
